cql3/expr/expression.cc

```diff
@@ -552,6 +581,15 @@ bool is_satisfied_by(const expression& restr, const column_value_eval_bag& bag)
                 });
             },
             [&] (const binary_operator& opr) { return is_satisfied_by(opr, bag); },
+            [] (const column_value&) -> bool {
+                throw std::runtime_error("A column cannot serve as a restriction by itself");
+            },
```

Comment:
btw, this can one day be lifted. If we have a column of type bool, SQL accepts `WHERE col`.